perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1620,6 +1622,15 @@ private String getSQLJoinFacility(UsersPageQuery query, MapSqlParameterSource na
 				namedParams.addValue("serviceId", query.getServiceId());
 				joinString += " join resource_services on resources.id=resource_services.resource_id and resource_services.service_id=(:serviceId)";
 			}
+
+			if (query.getConsentStatuses() != null && !query.getConsentStatuses().isEmpty()) {
+				List<String> statusStrings = query.getConsentStatuses().stream()
+					.map(ConsentStatus::toString).toList();
+				namedParams.addValue("consentStatuses", statusStrings);
+				joinString += " join consents on users.id = consents.user_id and (select consents.status from consents where consents.user_id=users.id" +
```

Comment:
Just a nitpick: this SQL query is quite complicated, and it is hard to read. Could you please reformat the query, so it looks something like this? 
![image](https://user-images.githubusercontent.com/48487536/195443512-c3ac6f3f-a8eb-4c38-ade0-27bc6f0aaf23.png)

Reply:
You're right, should have done this initially. My bad